packages/diagrams/lib/src/tabs.dart

```diff
@@ -0,0 +1,124 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:diagram_capture/diagram_capture.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'diagram_step.dart';
+
+const Duration _kTabScrollDuration = Duration(milliseconds: 300);
+const Duration _kPauseDuration = Duration(seconds: 1);
+final Duration _kTotalAnimationTime =
+    _kTabScrollDuration
+  + _kPauseDuration
+  + _kTabScrollDuration
+  + _kPauseDuration;
+const double _kAnimationFrameRate = 60.0;
+final List<GlobalKey> _tabKeys = <GlobalKey>[
+  GlobalKey(),
+  GlobalKey(),
+];
+
+class TabsDiagram extends StatefulWidget implements DiagramMetadata {
+  const TabsDiagram(this.name);
+
+  @override
+  final String name;
+
+  @override
+  State<TabsDiagram> createState() => TabsDiagramState();
+}
+
+class TabsDiagramState extends State<TabsDiagram> with SingleTickerProviderStateMixin {
+  final List<Tab> myTabs = <Tab>[
+    Tab(key: _tabKeys[0], text: 'Left'),
```

Comment:
This looks great, the animation is very subtle though. It may be worth adding something on the left and right pages that are different. This would make the change more apparent than the little navigation cues in the tab bar.

Reply:
I agree, I'll look into a way to highlight the tab change a little better.